src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -1194,6 +1195,32 @@ public function getRouteKeyName()
     }
 
     /**
+     * Get the translation name for the model.
+     *
+     * @return string
+     */
+    public function getTranslationName()
+    {
+        $key = $this->getTranslationNameKey();
+
+        if (Lang::has($key)) {
+            return Lang::get($key);
+        }
+
+        return Str::snake(class_basename($this), ' ');
+    }
+
+    /**
+     * Get the translation name key for the model.
+     *
+     * @return string
+     */
+    public function getTranslationNameKey()
+    {
+        return 'models.'.Str::snake(class_basename($this));
+    }
```

Comment:
hmmm, yea, perhaps a `Translatable` trait that the end user can `use` to their base class if they want the functionality...or something along those lines. I think that could work.

Having not done that myself, if you are using eloquent outside of Laravel, would you also be utilising the `routeKey` and those other helpers? Perhaps if you already get those even outside of Laravel this would be okay to tack on (if the community / Taylor are even keen for this functionality to be introduced)?